kornia/morphology/morphology.py

```diff
@@ -74,9 +80,13 @@ def dilation(
         neighborhood = structuring_element.clone()
         neighborhood[kernel == 0] = -max_val
 
-    output = output.unfold(2, se_h, 1).unfold(3, se_w, 1)
-    output, _ = torch.max(output + neighborhood.flip((0, 1)), 4)
-    output, _ = torch.max(output, 4)
+    B, C, H, W = tensor.size()
+    Hpad, Wpad = output.shape[-2:]
+    reshape_kernel = _neight2channels_like_kernel(kernel)
+    output, _ = F.conv2d(
+        output.view(B * C, 1, Hpad, Wpad), reshape_kernel, padding=0, bias=neighborhood.view(-1).flip(0)
```

Comment:
```suggestion
        output.view(B * C, 1, h_pad, w_pad), reshape_kernel, padding=0, bias=neighborhood.view(-1).flip(0)
```

Reply:
clone - because we are replacing the values in the next line. Without such thing, it fails if we try to backprop